app/uk/gov/hmrc/exports/models/declaration/ModeOfTransportCode.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.models.declaration
+
+import play.api.libs.json.{JsString, JsonValidationError, Reads, Writes}
+
+sealed abstract class ModeOfTransportCode(val value: String) {
+  def isDefined: Boolean = this != ModeOfTransportCode.Empty
+}
+
+object ModeOfTransportCode {
+
+  case object Maritime extends ModeOfTransportCode("1")
+  case object Rail extends ModeOfTransportCode("2")
+  case object Road extends ModeOfTransportCode("3")
+  case object Air extends ModeOfTransportCode("4")
+  case object PostalConsignment extends ModeOfTransportCode("5")
+  case object FixedTransportInstallations extends ModeOfTransportCode("7")
+  case object InlandWaterway extends ModeOfTransportCode("8")
+  case object Unknown extends ModeOfTransportCode("9")
+  case object Empty extends ModeOfTransportCode("")
+
+  def apply(code: String): ModeOfTransportCode = reverseLookup.getOrElse(code, Empty)
+
+  val allowedModeOfTransportCodes: Set[ModeOfTransportCode] =
```

Comment:
Do you need this "allowed: method?

Reply:
This is just for the sake of the next method.